qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -13,337 +13,200 @@
-        if qubit_state["XY plane"]:
-            prob_noise = 1 - (np.exp(-delay / t2hahn))
-            if self._rng.random() < prob_noise:
-                if self._rng.random() < 0.5:
-                    new_qubit_state = {
-                        "XY plane": False,
-                        "ZX plane": True,
-                        "Theta": 0,
-                    }
-                else:
-                    new_qubit_state = {
-                        "XY plane": False,
-                        "ZX plane": True,
-                        "Theta": np.pi,
-                    }
-            else:
-                phase = frequency * delay
-                new_theta = qubit_state["Theta"] + phase
-                new_theta = new_theta % (2 * np.pi)
-                new_qubit_state = {"XY plane": True, "ZX plane": False, "Theta": new_theta}
-        else:
-            if not isclose(qubit_state["Theta"], np.pi) and not isclose(qubit_state["Theta"], 0):
-                raise QiskitError("Delay gate supported only if the qubit is on the XY plane.")
-        return new_qubit_state
-
-    def _rx_gate(self, qubit_state: dict, angle: float) -> dict:
-        """
-        Apply Rx gate.
+    @classmethod
+    def _default_options(cls) -> Options:
+        return Options()
 
-        Args:
-            qubit_state(dict): The state of the qubit before operating the gate.
-            angle(float): The angle of the rotation.
+    def run(self, circuits, shots: int = 1024, **options) -> Job:
 
-        Returns:
-            dict: The state of the qubit after operating the gate.
+        passes = []
 
-        Raises:
-            QiskitError: if angle is not ±π/2 or ±π. Those are the only supported angles.
-        """
+        if isinstance(circuits, QuantumCircuit):
+            circuits = [circuits]
 
-        if qubit_state["XY plane"]:
-            if isclose(angle, np.pi):
-                new_theta = -qubit_state["Theta"]
-                new_theta = new_theta % (2 * np.pi)
-                new_qubit_state = {
-                    "XY plane": True,
-                    "ZX plane": False,
-                    "Theta": new_theta,
-                }
-            elif isclose(angle, np.pi / 2):
-                new_theta = (np.pi / 2) - qubit_state["Theta"]
-                new_theta = new_theta % (2 * np.pi)
-                new_qubit_state = {
-                    "XY plane": False,
-                    "ZX plane": True,
-                    "Theta": new_theta,
-                }
-            elif isclose(angle, -np.pi / 2):
-                new_theta = np.abs((-np.pi / 2) - qubit_state["Theta"])
-                new_theta = new_theta % (2 * np.pi)
-                new_qubit_state = {
-                    "XY plane": False,
-                    "ZX plane": True,
-                    "Theta": new_theta,
-                }
-            else:
+        for circuit in circuits:
+            if circuit.num_qubits != self.num_qubits:
```

Comment:
This test doesn't help too much for experiments because by the time an experiment gets to `run()` it has already done a transpile and if there are more qubits in the circuit than in the Target there would have been a transpile error.